elephant/spike_train_surrogates.py

```diff
@@ -550,6 +553,72 @@ def jitter_spikes(spiketrain, bin_size, n_surrogates=1):
             for s in surr]
 
 
+@deprecated_alias(binsize='bin_size', n='n_surrogates')
+def bin_shuffling(
+        binned_spiketrain, max_displacement, n_surrogates=1, sliding=False):
+    """
+    Bin shuffling surrogate generation.
+    Given a binned spiketrain, the function shuffles the binned spike
+    train entries inside windows of the twice the length of `max_displacement`,
+    either with sliding or exclusive windows.
+
+    Parameters
+    ----------
+    binned_spiketrain : conv.BinnedSpikeTrain
+        the binned spiketrain to create surrogates of
+    max_displacement : int
+        number of bins the a single spike can be displaced
+    n_surrogates : int, optional
+        number of surrogates
+        Default : 1
+    sliding : bool, optional
+        option to have a sliding window
+        Default : False
+
+    Returns
+    -------
+    binned_surrogates : list of conv.BinnedSpikeTrain
+        Each entry of the list is a binned surrogate spiketrain
+    """
+    displacement_window = 2 * max_displacement
+
+    binned_spiketrain_bool = binned_spiketrain.to_bool_array()[0]
+    st_length = len(binned_spiketrain_bool)
+
+    surrogate_spiketrains = []
+    for surrogate_id in range(n_surrogates):
+        surrogate_spiketrain = np.copy(binned_spiketrain_bool)
+        if sliding:
+            for window_position in range(st_length - displacement_window):
+                # shuffling the binned spike train within the window
+                np.random.shuffle(
+                    surrogate_spiketrain[
+                        window_position:
+                        window_position + displacement_window])
+        else:
+            windows = st_length // displacement_window
+            windows_remainder = st_length % displacement_window
+            # TODO: reshape the matrix and avoid for loop
```

Comment:
I don't understand how you're going to reshape the matrix and avoid the for loop
did you mean reshaping to the width of the displacement_window and then reshuffling by axis?
in any case, todo should be either dealt with or removed

Reply:
There had been the idea to reshuffle by axis, but since it's not the default option and thus we won't need it so often. I will just remove the TODO and not look in detail on how to solve it.